homeassistant/components/unifi_video/camera.py

```diff
@@ -0,0 +1,125 @@
+"""Support for Ubiquiti's Unifi Video cameras."""
+import logging
+
+from unifi_video import UnifiVideoAPI
+import voluptuous as vol
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_SSL
+from homeassistant.components.camera import Camera, PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_API_KEY = "api_key"
+CONF_VERIFY_CERT = "verify_cert"
+
+DEFAULT_PORT = 7443
+DEFAULT_SSL = True
+DEFAULT_VERIFY_CERT = True
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+        vol.Optional(CONF_VERIFY_CERT, default=DEFAULT_VERIFY_CERT): cv.boolean,
+    }
+)
+
+class UnifiVideoError(HomeAssistantError):
+    """General Unifi Video exception."""
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Discover cameras on a Unifi Video."""
+    host = config[CONF_HOST]
+    api_key = config[CONF_API_KEY]
+    port = config[CONF_PORT]
+    ssl = config[CONF_SSL]
+    verify_cert = config[CONF_VERIFY_CERT]
+
+    try:
+        uva = UnifiVideoAPI(
+            api_key=api_key,
+            addr=host,
+            port=port,
+            schema='https' if ssl else 'http',
+            verify_cert=verify_cert,
+            check_ufv_version=False,
+        )
+    except Exception as e:
+        raise UnifiVideoError(f"Unifi platform setup error: {e}")
+
+    _LOGGER.info(f"Connected to {uva._name} v{uva._version}")
+
+    add_entities(
+        [
+            UnifiVideoCamera(camera)
+            for camera in uva.cameras
+        ]
+    )
+
+
+class UnifiVideoCamera(Camera):
+    """A Ubiquiti Unifi Video Camera."""
+
+    def __init__(self, camera):
+        """Initialize a Unifi camera."""
+        super().__init__()
+        self._camera = camera
+        self.is_streaming = False
+    
+    def update(self):
+        """Update camera data."""
+        #FIXME: don't know what to do with that, periodic updates?
+        self._camera.update()
```

Comment:
After reading the docs and other platforms, I'm still not sure what's the best way to trigger an update of the state of the camera.
Making an API call as soon as a property is called seems overkill and is resulting in multiple API calls (at least 1 per camera) for no reason.